google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/TransactionRunnerImpl.java

```diff
@@ -321,96 +302,98 @@ public TransactionRunner allowNestedTransaction() {
     }
   }
 
-  private <T> T runInternal(TransactionCallable<T> callable) {
-    BackOff backoff = SpannerImpl.newBackOff();
-    final Context context = Context.current();
-    int attempt = 0;
-    // TODO: Change this to use TransactionManager.
-    while (true) {
-      checkState(
-          isValid, "TransactionRunner has been invalidated by a new operation on the session");
-      SpannerImpl.checkContext(context);
```

Comment:
What happens if the context is cancelled now? Are the same error messages thrown for deadline exceeded and cancelled?

Reply:
In the initial version of this PR, the context was ignored. I've added a check for context cancellation to the generic retry helper, and it will now throw the same exceptions as previous retry loops.